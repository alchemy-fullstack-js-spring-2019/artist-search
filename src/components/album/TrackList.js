import React from 'react';
import PropTypes from 'prop-types';
import Track from './Track';

function TrackList({ artist, tracks, lyricsLoad }) {
  const trackList = tracks.map(track => {
    return (
      <li key={track.id} >
        <Track artist={artist} songTitle={track.title} lyricsLoad={lyricsLoad} />
      </li>
    );
  });

  return (
    <ol>
      {trackList}
    </ol>
  );
}

TrackList.propTypes = {
  artist: PropTypes.string.isRequired,
  tracks: PropTypes.array.isRequired,
  lyricsLoad: PropTypes.func.isRequired
};

export default TrackList;
