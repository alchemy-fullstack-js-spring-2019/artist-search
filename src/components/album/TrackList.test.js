import React from 'react';
import { shallow } from 'enzyme';
import TrackList from './TrackList';

describe('TrackList component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<TrackList artist='' tracks={[]} lyricsLoad={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
