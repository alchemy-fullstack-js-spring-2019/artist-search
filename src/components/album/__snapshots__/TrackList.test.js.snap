// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackList component matches snapshot 1`] = `
<ol
  style={
    Object {
      "padding": "25px",
    }
  }
/>
`;
